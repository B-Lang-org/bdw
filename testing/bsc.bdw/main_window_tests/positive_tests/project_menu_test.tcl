
# Creates a new project
handlers::new_project new_project

# Displays the project status
commands::display_project_status

# Sets the compilation type to bsc
handlers::set_compilation_type bsc

# Sets the project editor to emacs
handlers::set_project_editor emacs

# Sets the bsc type and options
handlers::set_bsc_options -bluesim -options -v -info-dir . -rts-options fff

# Sets the link type to bsc
handlers::set_link_type bsc

# Sets the bsc type and options
handlers::set_link_bsc_options sim_exe -path mult4 -options -v

# Sets the bluesim options
handlers::set_bluesim_options -V

# Adds mult4 dir to search paths
handlers::set_search_paths mult4

# Sets the top file and module 
commands::set_top_module mult4/Mult4Tb.bsv mkMult4Tb

# Displays the project status
commands::display_project_status

# Sets the bsc type and options
handlers::set_bsc_options -verilog 

# Sets the verilog simulator
handlers::set_verilog_simulator modelsim -options -modelsim

# Displays the project status
commands::display_project_status

# Closes the project
handlers::close_project

# Displays the project status
commands::display_project_status

# Exits the workstation
exit
