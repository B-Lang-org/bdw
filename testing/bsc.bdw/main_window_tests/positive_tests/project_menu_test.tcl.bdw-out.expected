COMP_BDIR = .
COMP_BSC_OPTIONS = -keep-fires
COMP_BSC_TYPE = bluesim
COMP_INFO_DIR = .
COMP_SIMDIR = .
COMP_TYPE = bsc
COPY_FLAGS = 0
COMP_VDIR = .
DIR = .
EDITOR_NAME = emacs
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -keep-fires
LINK_OUTDIR = .
LINK_OUTNAME = out
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MAKE_CLEAN = clean
MAKE_FILE = Makefile
MAKE_FULLCLEAN = clean
MODULE_PLACEMENT = 550x350+50+500
NAME = new_project
PATHS = .
PATHS = Libraries
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = cvc
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = 0
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
COMP_BDIR = .
COMP_BSC_OPTIONS = -v
COMP_BSC_TYPE = bluesim
COMP_INFO_DIR = .
COMP_RTS_OPTIONS = fff
COMP_SIMDIR = .
COMP_TYPE = bsc
COPY_FLAGS = 0
COMP_VDIR = .
DIR = .
EDITOR_NAME = emacs
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -v
LINK_OUTDIR = mult4
LINK_OUTNAME = sim_exe
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MAKE_CLEAN = clean
MAKE_FILE = Makefile
MAKE_FULLCLEAN = clean
MODULE_PLACEMENT = 550x350+50+500
NAME = new_project
PATHS = .
PATHS = Libraries
PATHS = mult4
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
RUN_OPTIONS = -V
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = cvc
TOP_FILE = mult4/Mult4Tb.bsv
TOP_MODULE = mkMult4Tb
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = 0
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
Current version does not support the simulator 'modelsim'
COMP_BDIR = .
COMP_BSC_OPTIONS = -v
COMP_BSC_TYPE = verilog
COMP_INFO_DIR = .
COMP_RTS_OPTIONS = fff
COMP_SIMDIR = .
COMP_TYPE = bsc
COPY_FLAGS = 0
COMP_VDIR = .
DIR = .
EDITOR_NAME = emacs
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -v
LINK_OUTDIR = mult4
LINK_OUTNAME = sim_exe
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MAKE_CLEAN = clean
MAKE_FILE = Makefile
MAKE_FULLCLEAN = clean
MODULE_PLACEMENT = 550x350+50+500
NAME = new_project
PATHS = .
PATHS = Libraries
PATHS = mult4
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
RUN_OPTIONS = -V
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = cvc
TOP_FILE = mult4/Mult4Tb.bsv
TOP_MODULE = mkMult4Tb
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = 0
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
COMP_BSC_OPTIONS = -keep-fires
COMP_BSC_TYPE = bluesim
COMP_INFO_DIR = .
COMP_TYPE = bsc
COPY_FLAGS = 0
EDITOR_NAME = emacs
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -keep-fires
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MODULE_PLACEMENT = 550x350+50+500
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = cvc
STATUS = saved
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = close
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
